shell/common/thread_host.h

```diff
@@ -35,9 +49,14 @@ struct ThreadHost {
 
   ThreadHost& operator=(ThreadHost&&) = default;
 
-  ThreadHost(std::string name_prefix, uint64_t type_mask);
+  ThreadHost(std::string name_prefix,
+             uint64_t type_mask,
+             ThreadHostConfig configure_host = ThreadHostConfig());
```

Comment:
Oh, ThreadHostConfig can't be a const reference, because in the ThreadHost constructor declare I need move the ThreadConfig, ThreadConfig is in the HostConfig.

Reply:
If ThreadHostConfig is const reference, can't move its members.

Reply:
Right, but the move isn't helping you because you are copying the ThreadHostConfig at the call site.  Right now you are copying the whole data structure then moving out its components.   Instead you should be passing it by const ref and copying out its components.  The reason that is better is that better is it's less error prone and more efficient since it is only copying the things it uses instead of the whole struct.